client/packages/core/__tests__/src/store.test.js

```diff
@@ -470,3 +470,58 @@ test('ruleParams no-ops', () => {
 
   checkIndexIntegrity(newStore);
 });
+
+test('deepMerge', () => {
+  const gameId = uuid();
+  const gameStore = transact(
+    store,
+    instaml.transform(
+      { attrs: store.attrs },
+      tx.games[gameId].update({
+        state: {
+          score: 100,
+          playerStats: { health: 50, mana: 30, ambitions: { win: true } },
+          inventory: ['sword', 'potion'],
```

Comment:
What happens if you put `undefined` as an item in the array?

Reply:
Oo good question! 

Say we merge: 

```js
{ inventory: ['sword', undefined, 'potion'], }
```

**On the client**

Since we overwrite arrays, we copy the array over as is: 

```js
{ state { ... , inventory: ['sword', undefined, 'potion'], } }
```

`JSON.stringify(['sword', undefined, 'potion'])` produces: 

```js
['sword', null, 'potion'], }
```

So eventually it will become: 

```
{ state { ... , inventory: ['sword', null, 'potion'], } }
```

--- 

In this case I _think_ it's reasonable. Added a test to be explicit about this. Let me know!

Reply:
LGTM!